Frontend/ui-customer/src/Helpers/Utils.purs

```diff
@@ -765,4 +800,93 @@ formatFareType :: String -> String
+    SPECIAL_PICKUP -> 
+      { icon : "ny_ic_location_pin_white"
+      , text : if isLocalStageOn ConfirmingLocation then getString SPECIAL_PICKUP_ZONE else getString SPECIAL_PICKUP_ZONE_RIDE
+      , infoPopUpConfig : Just $ { title : getString SPECIAL_PICKUP_ZONE
+                                 , primaryText : getString WE_WILL_TRY_TO_CONNECT_YOU_WITH_DRIVER_IN_CLOSEST_PICKUP_ZONE
+                                 , secondaryText : getString THIS_PROVIDES_YOU_AN_INSTANT_PICKUP_EXPERIENCE
+                                 , primaryButtonText : getString GOT_IT
+                                 , icon : "ny_ic_sp_pickup_zone_map" }
+      , backgroundColor : Color.green900
+      }
+    AUTO_BLOCKED -> 
+      { icon : "ny_ic_zone_walk"
+      , text : getString GO_TO_SELECTED_PICKUP_SPOT_AS_AUTOS_ARE_RESTRICTED
+      , infoPopUpConfig : Nothing
+      , backgroundColor : Color.blue800
+      }
+    METRO ->
+      { icon : "ny_ic_metro_white"
+      , text : getString METRO_RIDE
+      , infoPopUpConfig : Nothing
+      , backgroundColor : Color.blue800
+      }
+    _ ->
+      { icon : "ny_ic_zone_walk"
+      , text : getString GO_TO_SELECTED_PICKUP_SPOT
+      , infoPopUpConfig : Nothing
+      , backgroundColor : Color.blue800
+      }
+
+zoneLabelIcon :: ZoneType -> String
+zoneLabelIcon zoneType =
+  case zoneType of
+    METRO -> "ny_ic_metro_white"
+    _ -> ""
+
+transformGeoJsonFeature :: Maybe String -> Array GateInfoFull -> String
+transformGeoJsonFeature geoJson gateInfoFulls =
+  if DS.null (fromMaybe "" geoJson) && DA.null gateInfoFulls then ""
+  else
+    AC.stringify $ AE.encodeJson CC.defaultGeoJson { features = geoJsonFeatures }
+    where
+      geoJsonFeatures :: Array CCT.GeoJsonFeature
+      geoJsonFeatures = 
+        DA.foldr (\(GateInfoFull gateInfoFull) specialZones -> 
+                  case gateInfoFull.geoJson of
+                    Just _ -> specialZones <> [createGeoJsonFeature (GateInfoFull gateInfoFull)]
+                    Nothing -> specialZones
+                 ) [] gateInfoFulls
+        <> case geoJson of
+              Just geoJson' -> DA.singleton CC.defaultGeoJsonFeature{ geometry = fromMaybe CC.defaultGeoJsonGeometry (decodeGeoJsonGeometry geoJson') }
+              Nothing -> []
+
+      decodeGeoJsonGeometry :: String -> Maybe CCT.GeoJsonGeometry
+      decodeGeoJsonGeometry stringGeometry =
+        case (AD.decodeJson =<< ADP.parseJson stringGeometry) of
+          Right resp -> Just resp
+          Left err   -> Nothing
+      
+      createGeoJsonFeature :: GateInfoFull -> CCT.GeoJsonFeature
+      createGeoJsonFeature (GateInfoFull gateInfoFull) = 
+        CC.defaultGeoJsonFeature {
+            properties {
+                name = gateInfoFull.name
+              , defaultDriverExtra = fromMaybe 0 gateInfoFull.defaultDriverExtra
+              , canQueueUpOnGate = fromMaybe false gateInfoFull.canQueueUpOnGate
+            }
+          , geometry = case gateInfoFull.geoJson of
```

Comment:
Can be simplified using. <$>
`fromMaybe defaultGeoJsonGeometry (decodeGeoJsonGeometry <$> gateInfoFull.geoJson)`